sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java

```diff
@@ -121,7 +119,59 @@ public static TaskInfo getTaskInfo(StateStore stateStore, TaskStatus taskStatus)
                     "Failed to find a task with TaskID: %s", taskStatus));
         }
 
-        return taskInfoOptional.get();
+        return taskInfoOptional.get().getName();
+    }
+
+    /**
+     * TaskInfo and TaskStatus objects referring to the same Task name are not written atomically.
+     * It is therefore possible for the states across these elements to become out of sync.  While the scheduler process
+     * is up they remain in sync.  This method guarantees produces an initial synchronized state.
+     *
+     * For example:
+     * <ol>
+     * <li>TaskInfo(name=foo, id=1) is written</li>
```

Comment:
Does the implementation make assumptions about ordering? Is it possible to have a `TaskStatus` written before a `TaskInfo`, for example?